file-formats/aifi/type/zif_aff_aifi_v1.intf.abap

```diff
@@ -0,0 +1,523 @@
+      "! <p class="shorttext">SOAP Method</p>
+      "! SOAP method
+      "! $required
+      soap_method                   TYPE zif_aff_types_v1=>ty_object_name_30,
+      " SOAP / web service record type is stored in /AIF/T_FINF-RECTYPERAW type /aif/rectyperaw
+      "! <p class="shorttext">SOAP Record Type</p>
+      "! SOAP record type
+      "! $required
+      soap_record_type              TYPE c LENGTH 45,
+      " /AIF/PROXY_XML_TRANSFORMATION
+      "! <p class="shorttext">Proxy XML Transformation</p>
+      "! Use proxy xml transformation
+      uses_proxy_xml_transformation TYPE abap_bool,
+      " /AIF/PRX_XXML
+      "! <p class="shorttext">Extend XML Handling</p>
+      "! Extend xml handling
+      uses_extended_xml_handling    TYPE abap_bool,
+    END OF ty_soap_settings.
+
+  TYPES:
+    "! <p class="shorttext">Event Settings</p>
+    "! Event settings
+    BEGIN OF ty_event_settings,
+      " EEE - Enterprise Event Engine (type /iwxbe/reg_repository_id)
+      "! <p class="shorttext">Repository ID</p>
+      "! Repository id
+      "! $required
+      repository_id        TYPE c LENGTH 10,
+      " EEE - Enterprise Event Engine (type /iwxbe/reg_id)
+      "! <p class="shorttext">Registration ID</p>
+      "! Registration id
+      "! $required
+      registration_id      TYPE c LENGTH 36,
+      " EEE - Enterprise Event Engine (type /iwxbe/reg_version)
+      "! <p class="shorttext">Registration Version</p>
+      "! Registration version
+      "! $required
+      registration_version TYPE n LENGTH 4,
+      " EEE - Enterprise Event Engine (type  /iwxbe/descr_topic)
+      "! <p class="shorttext">Topic</p>
+      "! Topic
+      "! $required
+      topic                TYPE c LENGTH 255,
+    END OF ty_event_settings.
+
+  TYPES:
+    "! <p class="shorttext">BgPF Settings</p>
+    "! Background processing framework settings
+    BEGIN OF ty_bgpf_settings,
+      " bgPF (type /aif/bgp_opid_e)
+      "! <p class="shorttext">Operation ID</p>
+      "! Operation id
+      "! $required
+      operation_id         TYPE zif_aff_types_v1=>ty_object_name_30,
+      " bgPF (type /aif/bgp_variant_id_e)
+      "! <p class="shorttext">Operation Variant ID</p>
+      "! Operation variant id
+      "! $required
+      operation_variant_id TYPE n LENGTH 10,
+      " bgPF (type /aif/bgp_variant_id_e)
+      "! <p class="shorttext">Operation Variant</p>
+      "! Operation variant
+      "! $required
+      operation_variant    TYPE string,
+    END OF ty_bgpf_settings.
+
+  TYPES:
+    "! <p class="shorttext">Application Interface</p>
+    "! Application interface
+    BEGIN OF ty_general_information,
+      "! <p class="shorttext">Integration Type</p>
+      "! Integration type
+      integration_type             TYPE c LENGTH 4,
+      "! <p class="shorttext">Integration Type</p>
+      "! Integration type description
+      integration_type_description TYPE c LENGTH 40,
```

Comment:
We do not want do declare the values on the UI/AFF, therefore enum is not desired. But I understand you proposal is to write the description to the INTEGRATION_TYPE instead of having an extra field. **I will check if we can do that and post another reply.**

Reply:
Adjustment is done and works. Commit will follow combined in case further adjustment is required.